Wine not detected on Gentoo

I am a user of Gentoo Linux, and _"somewhat recently" (in November, 2017)_, [the other Gentoo people decided to completely repackage wine](https://wiki.gentoo.org/wiki/Wine:_New_Packaging). Gone were the days of having to emerge _"app-emulation/wine"_, instead it's now a combination of _"virtual/wine"_ and _"app-emulation/wine-vanilla"_, _"app-emulation/wine-staging"_ or _"app-emulation/wine-d3d"_.

What does that mean?

That means that everything wine-related is found somewhere else than before, unless I manually symlink or something. 

Which in turn means this:

```
-- The C compiler identification is GNU 7.2.0
-- The CXX compiler identification is GNU 7.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.28") 
-- Found Git: /usr/bin/git (found version "2.13.6") 

Configuring LMMS
--------------------------
* Project version             : 1.2.0-rc5.9
*   Major version             : 1
*   Minor version             : 2
*   Release version           : 0
*   Stage version             : rc5
*   Build version             : 9
*

Optional Version Usage:
--------------------------
*   Override version:           -DFORCE_VERSION=x.x.x-x
*   Ignore Git information:     -DFORCE_VERSION=internal

PROCESSOR: x86_64
Machine: x86_64-pc-linux-gnu
-- Target host is 64 bit
-- Looking for include file stdint.h
-- Looking for include file stdint.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for include file semaphore.h
-- Looking for include file semaphore.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file sys/ipc.h
-- Looking for include file sys/ipc.h - found
-- Looking for include file sys/shm.h
-- Looking for include file sys/shm.h - found
-- Looking for include file sys/time.h
-- Looking for include file sys/time.h - found
-- Looking for include file sched.h
-- Looking for include file sched.h - found
-- Looking for include file sys/soundcard.h
-- Looking for include file sys/soundcard.h - found
-- Looking for include file soundcard.h
-- Looking for include file soundcard.h - not found
-- Looking for include file fcntl.h
-- Looking for include file fcntl.h - found
-- Looking for include file sys/ioctl.h
-- Looking for include file sys/ioctl.h - found
-- Looking for include file ctype.h
-- Looking for include file ctype.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file process.h
-- Looking for include file process.h - not found
-- Looking for include file locale.h
-- Looking for include file locale.h - found
-- Looking for Q_WS_X11
-- Looking for Q_WS_X11 - found
-- Looking for Q_WS_WIN
-- Looking for Q_WS_WIN - not found
-- Looking for Q_WS_QWS
-- Looking for Q_WS_QWS - not found
-- Looking for Q_WS_MAC
-- Looking for Q_WS_MAC - not found
-- Found Qt4: /usr/bin/qmake (found suitable version "4.8.7", minimum required is "4.6.0;COMPONENTS;QtCore;QtGui;QtXml") 
-- Found Qt translations in /usr/share/qt4/translations
-- Checking for module 'sndfile>=1.0.11'
--   Found sndfile, version 1.0.26
-- Checking for module 'carla-standalone>=1.9.5'
--   Found carla-standalone, version 1.9.5
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found SDL: /usr/lib64/libSDL.so;-lpthread (found version "1.2.15") 
-- Checking for module 'portaudio-2.0'
--   Found portaudio-2.0, version 19
-- Found Portaudio: portaudio;asound;m;pthread
-- Could NOT find SOUNDIO (missing:  SOUNDIO_LIBRARY SOUNDIO_INCLUDE_DIR) 
-- Checking for module 'libpulse'
--   Found libpulse, version 11.1
-- Found PulseAudio Simple: /usr/lib64/libpulse.so
-- Found Lame: /usr/include  
-- Looking for vorbis_bitrate_addblock in vorbis
-- Looking for vorbis_bitrate_addblock in vorbis - found
-- Found OggVorbis: /usr/lib64/libogg.so;/usr/lib64/libvorbis.so;/usr/lib64/libvorbisfile.so;/usr/lib64/libvorbisenc.so
-- Looking for snd_seq_create_simple_port in asound
-- Looking for snd_seq_create_simple_port in asound - found
-- Found ALSA: /usr/lib64/libasound.so
-- Looking for include file machine/soundcard.h
-- Looking for include file machine/soundcard.h - not found
-- Looking for include file linux/awe_voice.h
-- Looking for include file linux/awe_voice.h - not found
-- Looking for include file awe_voice.h
-- Looking for include file awe_voice.h - not found
-- Looking for include file /usr/src/sys/i386/isa/sound/awe_voice.h
-- Looking for include file /usr/src/sys/i386/isa/sound/awe_voice.h - not found
-- Looking for include file /usr/src/sys/gnu/i386/isa/sound/awe_voice.h
-- Looking for include file /usr/src/sys/gnu/i386/isa/sound/awe_voice.h - not found
-- Looking for C++ include sys/asoundlib.h
-- Looking for C++ include sys/asoundlib.h - found
-- Looking for C++ include alsa/asoundlib.h
-- Looking for C++ include alsa/asoundlib.h - found
-- Looking for snd_pcm_resume in asound
-- Looking for snd_pcm_resume in asound - found
-- Checking for module 'jack>=0.77'
--   Found jack, version 1.9.10
-- Checking for module 'fftw3f>=3.0.0'
--   Found fftw3f, version 3.3.4
-- Found OpenGL: /usr/lib64/libGL.so  
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib64/libX11.so
-- Found FLTK: /usr/lib64/fltk/libfltk_images.so;/usr/lib64/fltk/libfltk_forms.so;/usr/lib64/fltk/libfltk_gl.so;/usr/lib64/libGL.so;/usr/lib64/fltk/libfltk.so  
-- Checking for module 'fluidsynth>=1.0.7'
--   Found fluidsynth, version 1.1.6
-- Checking for module 'gig'
--   Found gig, version 4.0.0
-- Could NOT find Wine (missing:  WINE_LIBRARIES) 
-- Checking for module 'samplerate>=0.1.8'
--   Found samplerate, version 0.1.8
-- Setting build type to 'Release' as none was specified.
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.13") 

Installation Summary
--------------------
* Install Directory           : /usr/local

Supported audio interfaces
--------------------------
* ALSA                        : OK
* JACK                        : OK (weak linking enabled)
* OSS                         : OK
* Sndio                       : 
* PortAudio                   : OK
* libsoundio                  : not found, please install libsoundio if you require libsoundio support
* PulseAudio                  : OK
* SDL                         : OK

Supported MIDI interfaces
-------------------------
* ALSA                        : OK
* OSS                         : OK
* Sndio                       : 
* JACK                        : OK (weak linking enabled)
* WinMM                       : <not supported on this platform>
* AppleMidi                   : <not supported on this platform>

Supported file formats for project export
-----------------------------------------
* WAVE                        : OK
* OGG/VORBIS                  : OK
* MP3/Lame                    : OK

Optional plugins
----------------
* Carla Patchbay & Rack       : OK
* SoundFont2 player           : OK
* Stk Mallets                 : not found, please install libstk0-dev (or similar) ;if you require the Mallets instrument
* VST-instrument hoster       : not found, please install (lib)wine-dev (or similar) - 64 bit systems additionally need gcc-multilib and g++-multilib
* VST-effect hoster           : not found, please install (lib)wine-dev (or similar) - 64 bit systems additionally need gcc-multilib and g++-multilib
* CALF LADSPA plugins         : OK
* CAPS LADSPA plugins         : OK
* CMT LADSPA plugins          : OK
* TAP LADSPA plugins          : OK
* SWH LADSPA plugins          : OK
* GIG player                  : OK


-----------------------------------------------------------------
IMPORTANT:
after installing missing packages, remove CMakeCache.txt before
running cmake again!
-----------------------------------------------------------------



-- Configuring done
CMake Error at src/3rdparty/rpmalloc/CMakeLists.txt:3 (add_library):
  Cannot find source file:

    rpmalloc/rpmalloc/rpmalloc.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx


CMake Error: Cannot determine link language for target "rpmalloc".
CMake Error: CMake can not determine linker language for target: rpmalloc
```

I want VST support, which is precisely why I am willing to clone the git source and compile from there instead of emerging _"media-sound/lmms" (which doesn't have VST support)_. Will this be remedied eventually? Or do I have to manually symlink?

By the way, I have _"=app-emulation/wine-vanilla-2.0.3"_ and _"=app-emulation/wine-vanilla-2.20"_

Aaaaaand, error!

```
[ 83%] Generating moc_VstPlugin.cxx
[ 83%] Generating ../RemoteVstPlugin
x86_64-pc-linux-gnu-ld: Relocatable linking with relocations from format elf64-x86-64 (/usr/lib64/wine-vanilla-2.20/wine/libwinecrt0.a(exe_entry.o)) to format elf32-i386 (RemoteVstPlugin.gH5dp8.o) is not supported
winebuild: x86_64-pc-linux-gnu-ld failed with status 1
winegcc: winebuild failed
make[2]: *** [plugins/vst_base/CMakeFiles/vstbase.dir/build.make:63: plugins/RemoteVstPlugin] Error 2
make[1]: *** [CMakeFiles/Makefile2:8028: plugins/vst_base/CMakeFiles/vstbase.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
```

So tried this, seemed to work (thank you Metalhead33), but now getting 

`[ 66%] Building CXX object plugins/vst_base/CMakeFiles/vstbase.dir/VstPlugin.cpp.o
/mnt/stripe/media/lmmsbuild/lmms/plugins/vst_base/VstPlugin.cpp:41:33: fatal error: X11EmbedContainer.h: No such file or directory
 #  include "X11EmbedContainer.h"
                                 ^
compilation terminated.
make[2]: *** [plugins/vst_base/CMakeFiles/vstbase.dir/build.make:97: plugins/vst_base/CMakeFiles/vstbase.dir/VstPlugin.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:8072: plugins/vst_base/CMakeFiles/vstbase.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 66%] Building CXX object plugins/watsyn/CMakeFiles/watsyn.dir/moc_Watsyn.cpp.o
`

Not certain where X11EmbedContainer.h comes from, any idea? (Using Gentoo as well, relatively up to date system, and using Wine 2.0.3

--- a/cmake/modules/FindWine.cmake
+++ b/cmake/modules/FindWine.cmake
@@ -18,23 +18,23 @@ set(WINE_LIBRARIES ${WINE_LIBRARY} )
 
 # Handle wine linking problems
 EXEC_PROGRAM(${WINE_CXX} ARGS "-v -m32 /dev/zero" OUTPUT_VARIABLE WINEBUILD_OUTPUT)
+STRING(REPLACE " " ";" WINEBUILD_FLAGS "${WINEBUILD_OUTPUT}")
 
-# Debian systems 
-IF("${WINEBUILD_OUTPUT}" MATCHES ".*x86_64-linux-gnu/wine/libwinecrt0.a.*")
-	SET(WINE_LIBRARY_FIX "/usr/lib/i386-linux-gnu/" )
-# Fedora systems 
-ELSEIF("${WINEBUILD_OUTPUT}" MATCHES "/usr/lib/lib64/wine/libwinecrt0.a.*")
-	SET(WINE_LIBRARY_FIX "/usr/lib/i386/")
-# Wine stable
-ELSEIF("${WINEBUILD_OUTPUT}" MATCHES "/opt/wine-stable/lib64/wine/libwinecrt0.a.*")
-	SET(WINE_LIBRARY_FIX "/opt/wine-stable/lib/")
-# Wine development
-ELSEIF("${WINEBUILD_OUTPUT}" MATCHES "/opt/wine-devel/lib64/wine/libwinecrt0.a.*")
-	SET(WINE_LIBRARY_FIX "/opt/wine-devel/lib/")
-# Wine staging
-ELSEIF("${WINEBUILD_OUTPUT}" MATCHES "/opt/wine-staging/lib64/wine/libwinecrt0.a.*")
-	SET(WINE_LIBRARY_FIX "/opt/wine-staging/lib/")
-ENDIF()
+FOREACH(FLAG ${WINEBUILD_FLAGS})
+	IF("${FLAG}" MATCHES "libwinecrt0.a.*")
+		# Debian systems
+		STRING(REPLACE "/lib/x86_64-" "/lib/i386-" FLAG "${FLAG}")
+		# Fedora systems
+		STRING(REPLACE "/lib/lib64" "/lib/i386" FLAG "${FLAG}")
+		# Gentoo systems
+		STRING(REPLACE "/lib/wine-" "/lib32/wine-" FLAG "${FLAG}")
+		# WineHQ (/opt/wine-stable, /opt/wine-devel, /opt/wine-staging)
+		STRING(REPLACE "/lib64/wine/" "/lib/wine/" FLAG "${FLAG}")
+
+		STRING(REGEX REPLACE "/wine/libwinecrt0.a.*" "" WINE_LIBRARY_FIX "${FLAG}")
+		SET(WINE_LIBRARY_FIX "${WINE_LIBRARY_FIX}/")
+	ENDIF()
+ENDFOREACH()
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Wine DEFAULT_MSG WINE_LIBRARIES WINE_INCLUDE_DIRS)

